frontend/website/src/Style.re

```diff
@@ -54,60 +59,138 @@ module Typeface = {
-        url("/static/font/IBMPlexMono-Medium-Latin1.woff"),
-      ],
-      ~fontStyle=`normal,
-      ~fontWeight=`semiBold,
-      (),
-    );
-  let _ =
-    fontFace(
-      ~fontFamily="IBM Plex Mono",
-      ~src=[
-        url("/static/font/IBMPlexMono-SemiBold-Latin1.woff2"),
-        url("/static/font/IBMPlexMono-SemiBold-Latin1.woff"),
-      ],
-      ~fontStyle=`normal,
-      ~fontWeight=`bold,
-      (),
-    );
-
-  let ibmplexserif =
-    fontFamily(
-      fontFace(
-        ~fontFamily="IBM Plex Serif",
-        ~src=[
-          url("/static/font/IBMPlexSerif-Medium-Latin1.woff2"),
-          url("/static/font/IBMPlexSerif-Medium-Latin1.woff"),
+  // To prevent "flash of unstyled text" on some browsers (firefox), we need
+  // to do insane things to mitigate it. Even though the CSS working group
+  // created `font-display: block` for this purpose, Firefox chooses to not
+  // follow the standard "wait for 3seconds before showing fallback fonts."
+  //
+  // Instead we can base64 the woff and woff2 fonts and include those directly
+  // in our stylesheets. Now those browsers have no choice but to show us the
+  // font correctly.
+  //
+  // Scafolding code adapted from Bs-css Css.re.
+  module Loader = {
+    let string_of_fontWeight = x =>
+      switch (x) {
+      | `num(n) => string_of_int(n)
+      | `thin => "100"
+      | `extraLight => "200"
+      | `light => "300"
+      | `normal => "400"
+      | `medium => "500"
+      | `semiBold => "600"
+      | `bold => "700"
+      | `extraBold => "800"
+      | `black => "900"
+      | `lighter => "lighter"
+      | `bolder => "bolder"
+      | `initial => "initial"
+      | `inherit_ => "inherit"
+      | `unset => "unset"
+      };
+
+    let string_of_fontStyle =
+      fun
+      | `normal => "normal"
+      | `italic => "italic"
+      | `oblique => "oblique"
+      | `initial => "initial"
+      | `inherit_ => "inherit"
+      | `unset => "unset";
+
+    let genFontFace = (~fontFamily, ~src, ~fontStyle=?, ~fontWeight=?, ()) => {
+      let fontStyle =
+        Js.Option.map((. value) => string_of_fontStyle(value), fontStyle);
+
+      let src =
+        src
+        |> List.map(s => {
+             let ext = {
+               let arr = Js.String.split(".", s);
+               arr[Array.length(arr) - 1];
+             };
+             let b64 = Node.Fs.readFileSync("./" ++ s, `base64);
+             "url(\"data:font/"
+             ++ ext
+             ++ ";base64,"
+             ++ b64
+             ++ "\") format(\""
+             ++ ext
+             ++ "\")";
+           })
+        |> String.concat(", ");
+
+      let fontStyle =
+        Belt.Option.mapWithDefault(fontStyle, "", s => "font-style: " ++ s);
+      let fontWeight =
+        Belt.Option.mapWithDefault(fontWeight, "", w =>
+          "font-weight: " ++ string_of_fontWeight(w)
+        );
+      let asString = {j|@font-face {
+      font-family: $fontFamily;
+      src: $src;
```

Comment:
Is it worth adding support for format("woff") etc now that we control this?

Reply:
already done (see the string constructed there)